### ISSUE TITLE
[DOCUMENTATION] test errors while going through contribution guideline

### ISSUE THREAD

**Issue by Luthaf** [joaomcteixeira/taurenmd#41]
I am trying to follow https://taurenmd.readthedocs.io/en/latest/contributing.html#code-development, and I got a few errors while running tox:

- `ERROR:  py36: InterpreterNotFound: python3.6`: would there be a way to automatically check if python 3.6 is installed instead of erroring if it can not be found?
- error related to missing dependencies, they should perhaps be added to `tox.ini`

```
___________________________________________________________________________________ ERROR collecting tests/test_clis.py ____________________________________________________________________________________
ImportError while importing test module '/Users/guillaume/code/tests/taurenmd/tests/test_clis.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
.tox/py37/lib/python3.7/site-packages/_pytest/python.py:511: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
.tox/py37/lib/python3.7/site-packages/py/_path/local.py:701: in pyimport
    __import__(modname)
.tox/py37/lib/python3.7/site-packages/_pytest/assertion/rewrite.py:152: in exec_module
    exec(co, module.__dict__)
tests/test_clis.py:12: in <module>
    from taurenmd import __main__ as main_module
.tox/py37/lib/python3.7/site-packages/taurenmd/__main__.py:10: in <module>
    from taurenmd.cli import maincli
.tox/py37/lib/python3.7/site-packages/taurenmd/cli.py:86: in <module>
    from taurenmd import cli_distances as cli_dist
.tox/py37/lib/python3.7/site-packages/taurenmd/cli_distances.py:49: in <module>
    from taurenmd.libs import libcli, libio, libmda, libplot  # noqa: F401
.tox/py37/lib/python3.7/site-packages/taurenmd/libs/libmda.py:14: in <module>
    import MDAnalysis as mda
E   ModuleNotFoundError: No module named 'MDAnalysis'
__________________________________________________________________________________ ERROR collecting tests/test_libcalc.py __________________________________________________________________________________
ImportError while importing test module '/Users/guillaume/code/tests/taurenmd/tests/test_libcalc.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
.tox/py37/lib/python3.7/site-packages/_pytest/python.py:511: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
.tox/py37/lib/python3.7/site-packages/py/_path/local.py:701: in pyimport
    __import__(modname)
.tox/py37/lib/python3.7/site-packages/_pytest/assertion/rewrite.py:152: in exec_module
    exec(co, module.__dict__)
tests/test_libcalc.py:2: in <module>
    import MDAnalysis as mda
E   ModuleNotFoundError: No module named 'MDAnalysis'
__________________________________________________________________________________ ERROR collecting tests/test_libmda.py ___________________________________________________________________________________
ImportError while importing test module '/Users/guillaume/code/tests/taurenmd/tests/test_libmda.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
.tox/py37/lib/python3.7/site-packages/_pytest/python.py:511: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
.tox/py37/lib/python3.7/site-packages/py/_path/local.py:701: in pyimport
    __import__(modname)
.tox/py37/lib/python3.7/site-packages/_pytest/assertion/rewrite.py:152: in exec_module
    exec(co, module.__dict__)
tests/test_libmda.py:4: in <module>
    import MDAnalysis as mda
E   ModuleNotFoundError: No module named 'MDAnalysis'
__________________________________________________________________________________ ERROR collecting tests/test_libmdt.py ___________________________________________________________________________________
ImportError while importing test module '/Users/guillaume/code/tests/taurenmd/tests/test_libmdt.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
.tox/py37/lib/python3.7/site-packages/_pytest/python.py:511: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
.tox/py37/lib/python3.7/site-packages/py/_path/local.py:701: in pyimport
    __import__(modname)
.tox/py37/lib/python3.7/site-packages/_pytest/assertion/rewrite.py:152: in exec_module
    exec(co, module.__dict__)
tests/test_libmdt.py:2: in <module>
    import mdtraj as md
E   ModuleNotFoundError: No module named 'mdtraj'
```

**Comment by joaomcteixeira** [comment]
Hi @Luthaf 

* regarding the missing interpreter, could you please add the following line to the `tox.ini` and see if it maintains the same error?
  * `skip_missing_interpreters=true`, add it just after [here](https://github.com/joaomcteixeira/taurenmd/blob/acff0c01f02d9202bad312695796d42d4ae54d99/tox.ini#L10)
  * I have created a new conda environment to attempt to reproduce your error but I could not. `tox` always finds the interpreters.
  * Also notice that, despite the *missing interpreter* errors that are known to happen with `tox`, all the environments are tested later in `Travis-CI`.

* regarding the dependencies, `tox.ini` already has information to download all dependencies, and `tox-conda` should manage all that. Have you installed `tox-conda` as well?
  * also, please notice that development of `taurenmd` is bound to Anaconda, at least for now:

> ([from the documentation](https://taurenmd.readthedocs.io/en/latest/contributing.html#general-instructions))Though taurenmd can be used without relying on Anaconda to manage its dependencies, there are actually some dependencies that are only available in the Anaconda ecosystem. Therefore, the full development of taurenmd is bound to Anaconda until this situation is solved, you can read further on this on our installation page.

Please notice also that `taurenmd` current version is [`0.8.14`](https://github.com/joaomcteixeira/taurenmd/releases/tag/v0.8.14), and the one proposed in the article is [`0.8.9`](https://github.com/joaomcteixeira/taurenmd/releases/tag/v0.8.9), however, there have **not** been crucial changed to the `tox.ini` file during these updates. From the `master` branch you can see the changes in the `tox.ini` file since `0.8.9` with the following command:

```
git diff 40cf753dfb3afe0c3fcbc49696d0982314fad8f2 acff0c01f02d9202bad312695796d42d4ae54d99 tox.ini
```

Should the documentation be clarified in this regard?

**Comment by Luthaf** [comment]
> skip_missing_interpreters=true, add it just after here

This fixed the issue indeed.

> Have you installed tox-conda as well?

I was missing tox-conda, with it everything seems to work fine. I had seen the documentation regarding Anaconda, and I did not understand it as "you can **only** run tests and develop the code with Anaconda". If pip/PyPi is not supported, I think that's fine but should be made clearer. 

But I though that only OpenMM was not available on conda, right? What prevents you from changing tox.ini from 

```
deps =
    pytest
    pytest-travis-fold
    pytest-cov
    coverage
    bioplottemplates
    pyquaternion
conda_deps =
    mdanalysis
    mdtraj
    openmm
    matplotlib
```

to 

```
deps =
    pytest
    pytest-travis-fold
    pytest-cov
    coverage
    bioplottemplates
    pyquaternion
    mdanalysis
    mdtraj
    matplotlib
conda_deps =
    openmm
```

**Comment by joaomcteixeira** [comment]
refers to https://github.com/openjournals/joss-reviews/issues/2175

**Comment by joaomcteixeira** [comment]
Hi @Luthaf,
Please, have a look at #46, I have clarified the CONTRIBUTING instructions.

Regarding your question:

> But I though that only OpenMM was not available with pip, right? What prevents you from changing tox.ini from

It is just a matter of time, testing and CI becomes much faster if installing dependencies via anaconda. This is also stated in the current version of the documentation.

I wait for your comment before closing and merging the PR.

**Comment by Luthaf** [comment]
The changes in #46 looks better!

**Comment by joaomcteixeira** [comment]
Thanks for your input and help @Luthaf, I am merging to it master now. :wink:
Closed by #46
